src/schema/v2/me/interestsConnection.ts

```diff
@@ -0,0 +1,63 @@
+import { GraphQLFieldConfig, GraphQLInt } from "graphql"
+import { connectionFromArraySlice } from "graphql-relay"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { snakeCase } from "lodash"
+import { pageable } from "relay-cursor-paging"
+import { ResolverContext } from "types/graphql"
+import { createPageCursors } from "../fields/pagination"
+import {
+  userInterestCategoryEnum,
+  UserInterestConnection,
+  userInterestInterestTypeEnum,
+} from "../userInterests"
+
+export const InterestsConnection: GraphQLFieldConfig<void, ResolverContext> = {
+  type: UserInterestConnection,
+  args: pageable({
+    category: {
+      type: userInterestCategoryEnum,
+      description:
+        "UserInterest category to select. 'collected_before' or 'interested_in_collecting' category",
+    },
+    interestType: {
+      type: userInterestInterestTypeEnum,
+      description:
+        "UserInterest InterestType to select. 'Artist' or 'Gene' type",
+    },
+    page: { type: GraphQLInt },
+    size: { type: GraphQLInt },
+  }),
+  resolve: async (_, args, { meUserInterestsLoader }) => {
+    if (!meUserInterestsLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    const { page, size, offset, ...rest } = convertConnectionArgsToGravityArgs(
+      args
+    )
+
+    const gravityArgs = Object.keys(rest).reduce(
```

Comment:
#minor this is nice when there's a a big list of arguments, but here there's just one that needs this handling, `interestType`, so can just do it all in-line, no need for fanciness